### DISCUSSION TITLE
Maximum number of lines read from the file source?

### DISCUSSION THREAD

**Discussion by junf-liu** [lf-edge/ekuiper#3213] [Q&A]
I want to read a file, read only once, now the number of file lines is 10000, can only read about 5000, how should I adjust?

**Comment by ngjaying** [comment]
There is no limit for lines. Do you mean you want to read only 5000 lines?

**Reply by junf-liu** [reply]
> There is no limit for lines. Do you mean you want to read only 5000 lines?

No, I want to read all the data in the file, but I can only read about 5000 lines and the buffer will be full, and the file will have tens of thousands of data

**Reply by ngjaying** [reply]
Just told you there is no limit in eKuiper. If you do see the problem, describe your steps/configurations/rules to let developers recreate the problem easier.

**Reply by junf-liu** [reply]
By the way, what are the conditions for using the unnest function? Version 1.13.4-slim

**Reply by junf-liu** [reply]
> 顺便说一句，使用 unnest 函数的条件是什么呢？版本 1.13.4-slim

can t find the result from the unnest function

**Reply by junf-liu** [reply]
> Just told you there is no limit in eKuiper. If you do see the problem, describe your steps/configurations/rules to let developers recreate the problem easier.
this is  rule file 
[export rules.json](https://github.com/user-attachments/files/17111156/export.rules.json)
